src/engine/enginebuffer.cpp

```diff
@@ -62,6 +62,8 @@ constexpr SINT kSamplesPerFrame = 2; // Engine buffer uses Stereo frames only
 // Rate at which the playpos slider is updated
 constexpr int kPlaypositionUpdateRate = 15; // updates per second
 
+#define RUBBERBANDV3 (RUBBERBAND_API_MINOR_VERSION >= 2 && RUBBERBAND_API_MINOR_VERSION >= 7)
```

Comment:
```suggestion
#define RUBBERBANDV3 (RUBBERBAND_API_MINOR_VERSION >= 2 && RUBBERBAND_API_MINOR_VERSION >= 7)
```

Reply:
But it would be nice to hide the dependency here.

Reply:
I don't spot the difference in your suggestion?

Reply:
Agree, including the entirety of RubberbandStretcher.h just for `RUBBERBAND_API_MAJOR-/MINOR_VERSION` is quite a big dependency, but what other choice do we have?

Reply:
would a global const defined in enginebufferscalerubberband.h work?

Reply:
Not really... We need to differentiate at build time and the only way is by including the header. It would be nice if rubberband had a separate header that just includes the version, but it doesn't.

Reply:
My original point was that the EngineScaler should handle the dependency not EngineBuffer. 
Than EngieBuffer can use a member of EngineScaler to query about that.

Reply:
understood. 
> We need to differentiate at build time

That as well, thus my naive question about global const (set at compile time in the EngineScaler).

> Than EngieBuffer can use a member of EngineScaler to query about that.

and that, too. Though how is that supossed to work for the enum in the EngineBuffer header?

Reply:
> That as well, thus my naive question about global const (set at compile time in the EngineScaler).

The issue with globals consts is that they get to be evaluated after the preprocessor finished. Since we want to do conditional compilation using `#if`, we need this value sooner, so we need a `#define` for that then. `RubberBandStretcher.h` is a pretty lean header (90% is just comments) so I guess its not a big problem to include that a bunch of times.

Reply:
> Though how is that supossed to work for the enum in the EngineBuffer header?

My idea was to remove all the guards from the static code as if we are at a V3 build. If the engine Finer is not available the option need to be hidden from the GUI, that's all.

Reply:
Sure, but hiding it in the GUI seems more complicated than working with the guards in both EngineBuffer and EngineBufferScaleRubberBand.

Are you suggesting to first populate the keylock combobox with _all_ values, then (probably in 
[DlgPrefSound::loadSettings](https://github.com/mixxxdj/mixxx/blob/e1ee93311268101f06a761a4e258aa3b73a76a73/src/preferences/dialog/dlgprefsound.cpp#L499-L502)) somehow query EngineBuffer / EngineBufferScaleRubberBand to decide if we remove `RubberBand finer` from the combobox (and also reset the config to RubberBand (faster)) ?